tasks/scripts/containerd-integration

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+set -e -u -x
+
+export GOPATH=$PWD/gopath
+export PATH=$GOPATH/bin:$PATH
+
+source ci/tasks/scripts/docker-helpers.sh
+
+main() {
+  sanitize_cgroups
+  start_containerd
+  run_test
+}
+
+start_containerd() {
```

Comment:
here I tried to follow as closely as possible what we already do for `docker`:

https://github.com/concourse/ci/blob/e196f9c67c3508ac50e3409e1af180c11aa960c0/tasks/scripts/docker-helpers.sh#L64-L72

Reply:
so the other thing we do for docker is [make sure it stops](https://github.com/concourse/ci/blob/master/tasks/scripts/with-docker-compose#L31-L38) when the build is done -- since otherwise i think we can leak the `docker` process. I'm not totally sure how this black magic works, but should we maybe be careful about stopping `containerd`?

Reply:
That's a good point!

I think the big reason for stopping the whole thing there is that we run a
quite expensive setup - we're essentially running an entire Concourse
installation.

While I do not oppose cleaning things so that we end up with less resources
being left behind, I wonder if that's a pattern to persue or not even when the
resource utilization is not high, in the sense that I see as a nice to have the
ability to hijack and then play around w/ the tests having all set up already.

e.g., for *every* task where a daemon/process gets "background'ed", should we
always set up an exit trap to ensure that it gets killed?

I'm leaning towards "if it's super expensive, then, yeah", otherwise, "leave
there" as it can be convenient.

wdyt?

Reply:
Yeah, we were definitely encountering cases where workers were getting overloaded by those zombie `dockerd`s. I'm guessing that `containerd`, as it's being exercised here, will not be quite as expensive?

Reply:
> will not be quite as expensive?

yeahh, it'll be super cheap 😁  even if we leave containers "running", those containers will do nothing, as opposed to Concourse, which does all of its gc, scanning, scheduling, etc